libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/actionbutton/CameraModeButtonBinder.kt

```diff
@@ -0,0 +1,33 @@
+package com.mapbox.navigation.dropin.actionbutton
+
+import android.view.ViewGroup
+import androidx.lifecycle.viewModelScope
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.core.lifecycle.MapboxNavigationObserver
+import com.mapbox.navigation.dropin.camera.CameraModeButtonComponentContractImpl
+import com.mapbox.navigation.dropin.internal.extensions.reloadOnChange
+import com.mapbox.navigation.dropin.navigationview.NavigationViewContext
+import com.mapbox.navigation.ui.base.lifecycle.UIBinder
+import com.mapbox.navigation.ui.maps.internal.ui.CameraModeButtonComponent
+import com.mapbox.navigation.ui.maps.view.MapboxCameraModeButton
+
+@ExperimentalPreviewMapboxNavigationAPI
+internal class CameraModeButtonBinder(
+    private val context: NavigationViewContext
+) : UIBinder {
+
+    override fun bind(viewGroup: ViewGroup): MapboxNavigationObserver {
+        return reloadOnChange(context.styles.cameraModeButtonParams) { params ->
+            val button = MapboxCameraModeButton(viewGroup.context, null, 0, params.style)
+            viewGroup.removeAllViews()
+            viewGroup.addView(button, params.layoutParams)
+
+            CameraModeButtonComponent(button) {
+                CameraModeButtonComponentContractImpl(
+                    context.viewModel.viewModelScope,
+                    context.store
+                )
+            }
+        }
```

Comment:
Why not do the same as you are doing here https://github.com/mapbox/mapbox-navigation-android/pull/6490/files#diff-bfe82a9aa5902eb8b53e4b16729122d2707e7ee830fc65c37ea592410fcc09f1R16-R21

Reply:
Because UIComponents should not be responsible for creation/re-creation of the views. That's Binders responsibility.
The `RoutePreviewButtonComponent` has been wrongfully refactored some time ago to handle button re-creation (and in fact violate the above rule) when `MapboxExtendableButtonParams` was introduced. 
Since refactoring of the `RoutePreviewButtonComponent` is outside the scope of this PR, I've limited my changes to just the introduction of `InfoPanelRoutePreviewButtonBinder` that does nothing but return the component.
In a follow-up we should address this discrepancy and migrate view recreation logic from UIComponents to Binders.

Reply:
👍🏻